### PR TITLE
HEC-1061 Fetch CTUTR from CRN api

### PR DESCRIPTION
**Pull request by meghana-viswanath**
Ticket - https://jira.tools.tax.service.gov.uk/browse/HEC-1061

Add connector to `hec-applicant-frontend` to fetch CTUTR using a CRN from `hec` backend.

### CODE REVIEW

File: app/uk/gov/hmrc/hecapplicantfrontend/services/TaxCheckService.scala [hunk 1]
```diff
@@ -53,6 +53,8 @@ trait TaxCheckService {
     hc: HeaderCarrier
   ): EitherT[Future, Error, CTStatusResponse]
 
+  def getCtutr(crn: CRN)(implicit hc: HeaderCarrier): EitherT[Future, Error, CTUTRFromCRNResponse]
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/hecapplicantfrontend/services/TaxCheckService.scala:56]: the rest of the FE doesn't really need to know about `CTUTRFromCRNResponse` - we could simplify and just return `CTUTR` here. In the future then, we may change this more easily to `Option[CTUTR]` if we want to handle the case where no CTUTR is returned more specifically when we come to do the controller stuff.

Comment (meghana-viswanath) [reply app/uk/gov/hmrc/hecapplicantfrontend/services/TaxCheckService.scala:56]: yup, makes sense 👍

Comment (meghana-viswanath) [reply app/uk/gov/hmrc/hecapplicantfrontend/services/TaxCheckService.scala]: updated in commit - https://github.com/hmrc/hec-applicant-frontend/pull/33/commits/4841378c069f6c645a3926535ad034d07d69878f

**Review by priyankadhiman3006 — APPROVED** [review]

File: app/uk/gov/hmrc/hecapplicantfrontend/services/TaxCheckService.scala (hunk 1 above)
Comment (meghana-viswanath) [reply app/uk/gov/hmrc/hecapplicantfrontend/services/TaxCheckService.scala]: @AndyHWChung you happy with the changes?

**Review by AndyHWChung — APPROVED** [review]

File: app/uk/gov/hmrc/hecapplicantfrontend/services/TaxCheckService.scala (hunk 1 above)
Comment (AndyHWChung) [reply app/uk/gov/hmrc/hecapplicantfrontend/services/TaxCheckService.scala]: sorry, missed it! yeah, looks good :+1:

**Pull request merged**